librz/arch/isa/msp430/msp430_il.c

```diff
@@ -0,0 +1,433 @@
+	return SEQ4(
+		SETL("old_destination", get_destination(op, current_addr)),
+		SETL("result", LOGNOT(VARL("old_destination"))),
+		set_destination(op, VARL("result"), current_addr),
+		set_inv_flags("result", "old_destination"));
+}
+
+RzILOpEffect *rz_msp430_lift_adc_instr(RzAnalysis *analysis, const Msp430Instruction *op, ut64 current_addr, int instr_size) {
+	return BRANCH(
+		check_if_zero_carry(),
+		NOP(),
+		rz_msp430_lift_inc_instr(analysis, op, current_addr, instr_size));
+}
+
+RzILOpEffect *rz_msp430_lift_sbc_instr(RzAnalysis *analysis, const Msp430Instruction *op, ut64 current_addr, int instr_size) {
+	return BRANCH(
+		check_if_zero_carry(),
+		rz_msp430_lift_dec_instr(analysis, op, current_addr, instr_size),
+		NOP());
+}
+
+// ************************************* ------------------------ ******************************** //
+// ************************************* ------------------------ ******************************** //
+// ************************************* Jump Instructions Lifter ******************************** //
+// ************************************* ------------------------ ******************************** //
+// ************************************* ------------------------ ******************************** //
+
+RZ_OWN RzILOpEffect *rz_msp430_lift_jump_instr(RZ_NONNULL RzAnalysis *analysis, const Msp430Instruction *op, ut64 current_addr, int instr_size) {
+	RzILOpBitVector *abs_addr = abs_addr_from_rel_addr(U16(current_addr), op->jmp_addr);
+	RzILOpEffect *jmp = JMP(abs_addr);
+
+	// unconditional jump
+	if (op->jmp_cond == MSP430_JMP) {
+		return jmp;
+	}
+	// otherwise, construct the condition to make a guarded branch
+	RzILOpBool *jmp_cond = jmp_condition_constructors[op->jmp_cond](MSP430_GETR(MSP430_SR));
+
+	// if applicable, invert the condition efficiently by swapping THEN and ELSE in the RzIL branch
+	if (jmp_condition_triggers[op->jmp_cond] == JMP_THEN) {
+		return BRANCH(jmp_cond, jmp, NOP());
+	} else {
+		return BRANCH(jmp_cond, NOP(), jmp);
+	}
+}
+
+// ************************************* -------------- ******************************** //
+// ************************************* -------------- ******************************** //
+// ************************************* End of Lifters ******************************** //
+// ************************************* -------------- ******************************** //
+// ************************************* -------------- ******************************** //
+
+RZ_OWN RzILOpEffect *rz_msp430_dummy() {
+	RZ_LOG_ERROR("UNREACHABLE CONTROL FLOW");
+	return NOP();
+}
+
+RzILOpEffect *rz_msp430_lift_todo(RzAnalysis *analysis, const Msp430Instruction *op, ut64 current_addr, int instr_size) {
+	NOT_IMPLEMENTED;
+}
+
+#include "rz_il/rz_il_opbuilder_end.h"
+
+static const MSP430InstructionLifter one_op_lifters[] = {
+	[MSP430_RRC] = rz_msp430_lift_rrc_instr,
+	[MSP430_SWPB] = rz_msp430_lift_swpb_instr,
+	[MSP430_RRA] = rz_msp430_lift_todo, // TODO
```

Comment:
```suggestion
	[MSP430_RRA] = rz_msp430_lift_todo,
```

Also below

Reply:
Also, this syntax is C20 standard, right? Could conflict with some compilers. cc @XVilka